pkg/daemon/daemon.go

```diff
@@ -573,7 +573,8 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 			}
 		}
 		reqDrain = reqDrain || systemdConfModified
-		reqReboot = reqReboot || systemdConfModified
+        // require reboot if drain needed for systemd mode
+		reqReboot = reqReboot || systemdConfModified || reqReboot
```

Comment:
It looks like the behaviour of the new line is exact same as of the old one. Do we need this change?

Reply:
@ykulazhenkov it's a typo, I'll update it

Reply:
nit: you can drop `systemdConfModified` from reqReboot condition

Reply:
> nit: you can drop `systemdConfModified` from reqReboot condition

I would like to make it more explicit for this particular case. That's why I left `systemdConfModified`